c++/src/kj/compat/http.c++

```diff
@@ -7110,8 +7110,14 @@ private:
                 // period and then abort the connection.
 
                 auto dummy = kj::heap<HttpDiscardingEntityWriter>();
-                auto lengthGrace = body->pumpTo(*dummy, server.settings.canceledUploadGraceBytes)
-                    .then([this](size_t amount) {
+                auto lengthGrace = kj::evalNow([&]() {
+                  return body->pumpTo(*dummy, server.settings.canceledUploadGraceBytes);
+                }).catch_([](kj::Exception&& e) -> size_t {
+                  // Reading from the input failed in some way. This may actually be the whole
+                  // reason we got here in the first place so don't propagate this error, just
+                  // give up on discarding the input.
```

Comment:
I think I've seen some mysteries (like errors causing other errors) over the years that could be explained by this.

Reply:
Though this might have introduced some test failures related to body-discarding: https://github.com/capnproto/capnproto/actions/runs/4079783138/jobs/7031513983#step:4:2429

Reply:
It would appear I failed to actually run Cap'n Proto's tests while developing this. :facepalm: